src/utils.ts

```diff
@@ -222,23 +240,19 @@ function parseJSONRPCMessage<T extends JSONValue>(
  * @param {any} [id] - Optional id for the error object in the RPC response.
  * @returns {JSONValue} The serialized ErrorRPC instance.
  */
-function fromError(error: ErrorRPC<any>, id?: any): JSONValue {
-  const data: { [key: string]: JSONValue } = {
-    errorCode: error.code,
-    message: error.message,
-    data: error.data,
-    type: error.constructor.name,
-  };
-  return {
-    error: {
-      data,
-    },
+function fromError(errorin: rpcErrors.ErrorRPCProtocol<any>, id?: any): JSONValue {
+  const error: { [key: string]: JSONValue } = {
+    errorCode: errorin.code,
+    message: errorin.message,
+    data: errorin.data,
+    type: errorin.constructor.name,
   };
+  return error;
 }
```

Comment:
This doesn't make sense to me. I think I need an explanation on what's going on with the `toError` and `FromError` and how that all works now.  Looking over it all it doesn't seem to make sense.

I'm not clear on what all the requirements are for that now.